src/ruby/pb/test/xds_client.rb

```diff
@@ -95,17 +96,45 @@ def get_client_stats(req, _call)
       end
       $watchers.delete_at($watchers.index(watcher))
     end
+    # convert results into proper proto object
+    rpcs_by_method = {}
+    watcher['rpcs_by_method'].each do | rpc_name, rpcs_by_peer |
+      rpcs_by_method[rpc_name] = LoadBalancerStatsResponse::RpcsByPeer.new(
+        rpcs_by_peer: rpcs_by_peer
+      )
+    end
     LoadBalancerStatsResponse.new(
+      rpcs_by_method: rpcs_by_method,
       rpcs_by_peer: watcher['rpcs_by_peer'],
       num_failures: watcher['no_remote_peer'] + watcher['rpcs_needed']
     );
   end
 end
 
+# execute 1 RPC and return remote hostname
+def execute_rpc(op, fail_on_failed_rpcs)
+  remote_peer = ""
```

Comment:
Question about the spec -- are the keys in `rpcs_by_peer` supposed to be hostnames, or socket addresses?

If socket addresses, we can use the `peer` getter on `op`s.

Reply:
Similar response to above - according to the new updated test spec, we need to expect this `hostname` key in the call's initial metadata.